src/account/default.ts

```diff
@@ -73,9 +73,12 @@ export class Account extends Provider implements AccountInterface {
     address: string,
     pkOrSigner: Uint8Array | string | SignerInterface,
     cairoVersion?: CairoVersion,
-    transactionVersion: ETransactionVersion.V2 | ETransactionVersion.V3 = ETransactionVersion.V2 // TODO: Discuss this, set to v2 for backward compatibility
+    transactionVersion:
+      | ETransactionVersion.V2
+      | ETransactionVersion.V3 = DEFAULT_TRANSACTION_VERSION,
+    instance?: boolean
```

Comment:
This `instance` parameter and the new `AccountInstance` class feels like it could be confusing to users. I suggest evaluating if a different term might be a better fit, maybe something like `inheritable` or `inheritableProvider`.

Adding @ivpavici since he's had success with naming things.